apisix/consumer.lua

```diff
@@ -101,10 +104,19 @@ local function plugin_consumer()
                 if not plugins[name] then
                     plugins[name] = {
                         nodes = {},
+                        len = 0,
                         conf_version = consumers.conf_version
                     }
                 end
 
+                local cached_consumer = consumers_id_cache[val.value.id]
```

Comment:
I do not know why we have to add `consumers_id_cache `

it seems useless

Reply:
The purpose of using consumer_id_cache is to avoid unnecessary table copies. I will refactor this part of the code to replace the table with an LRU cache.